Sources/Yams/Parser.swift

```diff
@@ -272,21 +272,35 @@ private extension Parser {
     func loadDocument() throws -> Node {
         let node = try loadNode(from: parse())
         try parse() // Drop YAML_DOCUMENT_END_EVENT
-        return node
+        return try node.resolvingAliases(withAnchors: self.anchors)
     }
 
     func loadNode(from event: Event) throws -> Node {
         switch event.type {
         case YAML_ALIAS_EVENT:
-            return try loadAlias(from: event)
+            do {
+                return try loadAlias(from: event)
```

Comment:
can we change the `loadAlias` to do `return Node(unresolved: alias, error: yamlError)` when an alias is not found in anchor array? instead of throw an exception in loadAlias and then catch it in loadNode and return ` Node(unresolved: alias, error: yamlError)`